### PR TITLE
List all config params in README.md

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
...
+
+- loginButtonName - the name as displayed on the login screen which is used to redirect to the IdP
+- autoRedirectOnLoginPage - if set to true the login page will redirect to the Idp right away
+- provider-url - the url where the IdP is living. In some cases (KeyCloak, Azure AD) this holds more then just a domain but also a path
+- client-id & client-secret - self-explanatory
+- scopes - depending on the IdP setup and needs the list of required scopes is entered here
+- insecure - no ssl verification will take place when talking to the IdP - DON'T use in production
+- provider-params - additional config depending on the IdP is to be entered here - usually only necessary if the IdP does not support service discovery
+- auth-params - additional parameters which are sent to the IdP during the auth requests
+- redirect-utl - the full url under which the ownCloud OpenId Connect redirect url is reachable - only needed in special setups
```
Comment (pmaier1) [inline README.md:64]: typo "u**t**l"

**Comment by mmattel** [comment]
I remember that there is an open docs PR. This info should maybe added to this PR.

**Comment by pmaier1** [comment]
> I remember that there is an open docs PR. This info should maybe added to this PR.

That's the plan, don't worry :)

**Review by pmaier1 — APPROVED** [review]
thanks a lot :heart:

File: README.md [hunk 2]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
     ]
   ]
 ];
+```
 
+#### All Configuration Values explained
+
+- loginButtonName - the name as displayed on the login screen which is used to redirect to the IdP
+- autoRedirectOnLoginPage - if set to true the login page will redirect to the Idp right away
+- provider-url - the url where the IdP is living. In some cases (KeyCloak, Azure AD) this holds more then just a domain but also a path
```
Comment (phil-davis) [inline README.md:58]:
```suggestion
- provider-url - the url where the IdP is living. In some cases (KeyCloak, Azure AD) this holds more than just a domain but also a path
```

File: README.md [hunk 3]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
...
+- client-id & client-secret - self-explanatory
+- scopes - depending on the IdP setup and needs the list of required scopes is entered here
```
Comment (phil-davis) [inline README.md:60]:
```suggestion
- scopes - depending on the IdP setup, needs the list of required scopes to be entered here
```
(should there be example(s) somewhere?)

File: README.md [hunk 4]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
...
+- insecure - no ssl verification will take place when talking to the IdP - DON'T use in production
```
Comment (phil-davis) [inline README.md:61]: set to what? boolean true, string "true" or?

File: README.md [hunk 5]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
...
+- provider-params - additional config depending on the IdP is to be entered here - usually only necessary if the IdP does not support service discovery
+- auth-params - additional parameters which are sent to the IdP during the auth requests
+- redirect-url - the full url under which the ownCloud OpenId Connect redirect url is reachable - only needed in special setups
+- use-token-introspection-endpoint - if set to true the token introspection endpoint is used to verify a given access token - only needed if the access token is not a JWT
+- token-introspection-endpoint-client-id & token-introspection-endpoint-client-secret - client id and secret to be used with the token introspection endpoint
+- post_logout_redirect_uri - a given url where the IdP should redirect to after logout
+- mode - the mode to search for user in ownCloud - either userid or email
+- search-attribute - the attribute which is taken from the access token JWT or user info endpoint to identify the user (names as claim as well)
```
Comment (phil-davis) [inline README.md:69]: `(names as claim as well)`
I don't understand what this means.

File: README.md [hunk 6]
```diff
@@ -49,9 +49,27 @@ $CONFIG = [
...
+- allowed-user-backends - limit the user which are allowed to login to a specific user backend - e.g. LDAP
```
Comment (phil-davis) [inline README.md:70]:
```suggestion
- allowed-user-backends - limit the users which are allowed to login to a specific user backend - e.g. LDAP
```

**Review by phil-davis — COMMENTED** [review]
Just a couple of comments. I guess this is only supposed to be a summary of what options exist, so there are not detailed examples.

**Comment by mmattel** [comment]
This is a new config parameter which should/must be documented also in core/`config.sample.php`
Else this would be a missing information! It would be good to add that soon, because there are already new parameters added and a upcoming `config-to-docs` run would transfer this into the documentation
@pmaier1

**Pull request merged**